src/apps/filters/h3.c

```diff
@@ -140,35 +185,177 @@ H3Error cellToBoundaryCmd(int argc, char *argv[]) {
+    DEFINE_CELL_ARG(cell, cellArg);
+    Arg *args[] = {&getBaseCellNumberArg, &helpArg, &cellArg};
+    PARSE_SUBCOMMAND(argc, argv, args);
+    // TODO: Should there be a general `isValidIndex`?
+    H3Error cellErr = H3_EXPORT(isValidCell)(cell);
+    H3Error edgeErr = H3_EXPORT(isValidDirectedEdge)(cell);
+    H3Error vertErr = H3_EXPORT(isValidVertex)(cell);
+    if (cellErr && edgeErr && vertErr) {
+        // Not exactly sure how to handle this
+        return cellErr | edgeErr | vertErr;
     }
-    if (has("cellToLatLng", 1, argv)) {
-        return cellToLatLngCmd(argc, argv);
+    // If we got here, we can use `getResolution` safely, as this is one of the
+    // few functions that doesn't do any error handling (for some reason? I
+    // don't see how this would ever be in a hot loop anywhere.
+    int baseCell = H3_EXPORT(getBaseCellNumber)(cell);
+    printf("%i", baseCell);
+    return E_SUCCESS;
+}
+
+SUBCOMMAND(stringToInt, "Converts an H3 index in string form to integer form") {
+    char *rawCell = calloc(16, sizeof(char));
+    Arg rawCellArg = {.names = {"-c", "--cell"},
+                      .required = true,
+                      .scanFormat = "%s",
+                      .valueName = "cell",
+                      .value = rawCell,
+                      .helpText = "H3 Cell Index"};
+    Arg *args[] = {&stringToIntArg, &helpArg, &rawCellArg};
+    PARSE_SUBCOMMAND(argc, argv, args);
+    H3Index c;
+    H3Error err = H3_EXPORT(stringToH3)(rawCell, &c);
+    if (err) {
+        free(rawCell);
+        return err;
     }
-    if (has("latLngToCell", 1, argv)) {
-        return latLngToCellCmd(argc, argv);
+    printf("%" PRIu64, c);
+    free(rawCell);
+    return E_SUCCESS;
+}
+
+SUBCOMMAND(intToString, "Converts an H3 index in int form to string form") {
+    H3Index rawCell;
+    Arg rawCellArg = {.names = {"-c", "--cell"},
+                      .required = true,
+                      .scanFormat = "%" PRIu64,
+                      .valueName = "cell",
+                      .value = &rawCell,
+                      .helpText = "H3 Cell Index"};
+    Arg *args[] = {&intToStringArg, &helpArg, &rawCellArg};
+    PARSE_SUBCOMMAND(argc, argv, args);
+    h3Println(rawCell);
+    return E_SUCCESS;
+}
+
+SUBCOMMAND(isValidCell, "Checks if the provided H3 index is actually valid") {
+    DEFINE_CELL_ARG(cell, cellArg);
+    Arg *args[] = {&isValidCellArg, &helpArg, &cellArg};
+    PARSE_SUBCOMMAND(argc, argv, args);
+    bool isValid = H3_EXPORT(isValidCell)(cell);
+    printf("%s", isValid ? "true" : "false");
+    return E_SUCCESS;
+}
+
+SUBCOMMAND(isResClassIII,
+           "Checks if the provided H3 index has a Class III orientation") {
+    DEFINE_CELL_ARG(cell, cellArg);
+    Arg *args[] = {&isResClassIIIArg, &helpArg, &cellArg};
+    PARSE_SUBCOMMAND(argc, argv, args);
+    bool isClassIII = H3_EXPORT(isResClassIII)(cell);
+    printf("%s", isClassIII ? "true" : "false");
+    return E_SUCCESS;
+}
```

Comment:
it's surprising that getResolution does a lot of extra checks but isResClassIII doesn't

Reply:
Yeah, I will update that.